### PR TITLE
Add utility function to VolumeData

### PR DESCRIPTION
**Pull request by nilsvu**
## Proposed changes

Adds a utility function to find the offset into the contiguous dataset stored in `h5::VolumeData`
for a particular `grid_name`.

### Types of changes:

- [ ] Bugfix
- [x] New feature
- [ ] Refactor

### Component:

- [x] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests and `clang-tidy`.
  For instructions on how to perform the CI checks locally refer to the [Dev
  guide on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

File: src/IO/H5/VolumeData.cpp
Comment (nilsdeppe) [inline src/IO/H5/VolumeData.cpp]: we should use `static_cast<size_t>(0)` in these cases to avoid implicit conversion, or (easier to read), include `Utilities/Literals.hpp` which allows you to do `0_st`

File: src/IO/H5/VolumeData.cpp
Comment (nilsdeppe) [inline src/IO/H5/VolumeData.cpp]: If you include `Utilities/Algorithm.hpp` you can do `alg::find(all_grid_names, grid_name)`, which is easier to read :)

File: src/IO/H5/VolumeData.cpp
Comment (nilsdeppe) [inline src/IO/H5/VolumeData.cpp]: can `element_data_offset` be const?

File: src/IO/H5/VolumeData.cpp
Comment (nilsdeppe) [inline src/IO/H5/VolumeData.cpp]: can `element_data_length` be const?

**Review by nilsdeppe — CHANGES_REQUESTED** [review]
I'm happy for you to squash

**Comment by nilsvu** [comment]
@nilsdeppe I addressed your comments and squashed. This needs a second review, perhaps @isaaclegred could you take a look?

File: src/IO/H5/VolumeData.hpp
Comment (isaaclegred) [inline src/IO/H5/VolumeData.hpp:163]: Will it ever make sense to call this function and not pass `get_grid_names` and `get_extents` to the `all_grid_names` and `all_extents` slots?  I guess I would be a bit worried that people may interpret this functionality as the ability to pass in a subset of grid names and extents and get the relative position or something.  Maybe you could add a comment about why you're explicitly passing the arguments rather than getting them inside the function?

File: src/IO/H5/VolumeData.hpp
Comment (isaaclegred) [inline src/IO/H5/VolumeData.hpp]: "Find the offset ... `grid_name`, and the number of points on `grid_name` ", or something similar

**Review by isaaclegred — CHANGES_REQUESTED** [review]
Just one minor thing and one question, feel free to squash.  Let me know if anything was unclear!

File: src/IO/H5/VolumeData.hpp
Comment (nilsvu) [reply src/IO/H5/VolumeData.hpp]: changed the docs to reflect that both the offset and the length is returned, please take a look

File: src/IO/H5/VolumeData.hpp
Comment (nilsvu) [reply src/IO/H5/VolumeData.hpp:163]: This function will be called many times (for each grid_name), but we don't need to re-compute the all_grid_names and all_extents every time. Added a comment to the docs, please take a look.

**Comment by kidder** [comment]
@isaaclegred @nilsdeppe this has been updated and squashed

**Review by isaaclegred — APPROVED** [review]
Looks good to me

**Comment by nilsvu** [comment]
@nilsdeppe please take a look. Thanks for reviewing @isaaclegred and @nilsdeppe :)

**Review by nilsdeppe — APPROVED** [review]

**Pull request merged**
